rfd/0092-error-message-guidelines.md

```diff
@@ -0,0 +1,552 @@
+2022-07-28T01:46:21Z DEBU [PROC:1]    Failed to connect to Auth Server directly. auth-addrs:[0.0.0.0:3025] error:[
+ERROR REPORT:
+Original Error: *trace.ConnectionProblemError connection error: desc = &#34;transport: Error while dialing failed to dial: dial tcp 0.0.0.0:3025: i/o timeout&#34;
+Stack Trace:
+        /go/src/github.com/gravitational/teleport/api/client/client.go:2758 github.com/gravitational/teleport/api/client.(*Client).GetDomainName
+        /go/src/github.com/gravitational/teleport/lib/auth/httpfallback.go:35 github.com/gravitational/teleport/lib/auth.(*Client).GetDomainName
+        /go/src/github.com/gravitational/teleport/lib/service/connect.go:1160 github.com/gravitational/teleport/lib/service.(*TeleportProcess).newClientDirect
+        /go/src/github.com/gravitational/teleport/lib/service/connect.go:1053 github.com/gravitational/teleport/lib/service.(*TeleportProcess).newClient
+        /go/src/github.com/gravitational/teleport/lib/service/connect.go:626 github.com/gravitational/teleport/lib/service.(*TeleportProcess).firstTimeConnect
+        /go/src/github.com/gravitational/teleport/lib/service/connect.go:235 github.com/gravitational/teleport/lib/service.(*TeleportProcess).connect
+        /go/src/github.com/gravitational/teleport/lib/service/connect.go:198 github.com/gravitational/teleport/lib/service.(*TeleportProcess).connectToAuthService
+        /go/src/github.com/gravitational/teleport/lib/service/connect.go:70 github.com/gravitational/teleport/lib/service.(*TeleportProcess).reconnectToAuthService
+        /go/src/github.com/gravitational/teleport/lib/service/service.go:2412 github.com/gravitational/teleport/lib/service.(*TeleportProcess).registerWithAuthServer.func1
+        /go/src/github.com/gravitational/teleport/lib/service/supervisor.go:521 github.com/gravitational/teleport/lib/service.(*LocalService).Serve
+        /go/src/github.com/gravitational/teleport/lib/service/supervisor.go:269 github.com/gravitational/teleport/lib/service.(*LocalSupervisor).serve.func1
+        /opt/go/src/runtime/asm_arm64.s:1263 runtime.goexit
+User Message: connection error: desc = &#34;transport: Error while dialing failed to dial: dial tcp 0.0.0.0:3025: i/o timeout&#34;] pid:2002.1 service/connect.go:1076
+```
+
+The auth service had an advertise addr of 0.0.0.0.
+
+Either the auth service itself or the proxy service could recognize that 0.0.0.0 is probably not what is correct.
+
+---
+> 📝 **Suggestion**
+> 
+> Craft a single line error message that indicates the source of the endpoint being used, include the network error, and include the go source file and line.
+> 
+> `Error while dialing to the advertise_ip for the auth_service: failed to dial: dial tcp 0.0.0.0:3025: i/o timeout pid:2002.1 service/connect.go:1076`
+> 
+---
+
+
+### Context Deadline Exceeded: proxy server hits maximum allowed connections
```

Comment:
These examples are great. I'd like to fix each of them.

I don't think getting this RFD approved and having new guidelines will magically fix these specific instances, so I'd recommend creating an issue for each of these with the relevant details.